uportal-war/src/main/java/org/apereo/portal/security/provider/cas/CasAssertionSecurityContext.java

```diff
@@ -44,16 +47,46 @@
 public class CasAssertionSecurityContext extends ChainingSecurityContext implements ICasSecurityContext {
     private static final String SESSION_ADDITIONAL_DESCRIPTORS_BEAN = "sessionScopeAdditionalDescriptors";
     private static final String CAS_COPY_ASSERT_ATTR_TO_USER_ATTR_BEAN = "casCopyAssertionAttributesToUserAttributes";
+    private static final String DECRYPT_CRED_TO_PWD = "decryptCredentialToPassword";
+    private static final String DECRYPT_CRED_TO_PWD_KEY = "decryptCredentialToPasswordPrivateKey";
+    private static final String CREDENTIAL_KEY = "credential";  // encrypted password attribute from CAS
+    private static final String PASSWORD_KEY = "password";  // user attribute expected by portlets
 
     // UP-4212 Transient because security contexts are serialized into HTTP Session (and webflow).
     private transient ApplicationContext applicationContext;
     private Assertion assertion;
     private boolean copyAssertionAttributesToUserAttributes = false;
+    private boolean decryptCredentialToPassword = false;
+    private String decryptCredentialToPasswordPrivateKey="/etc/cas/private.p8";
+    private static PrivateKey key = null;
```

Comment:
Yes. Loading the key requires disk IO. However, your comment made me realize that they key should be loaded during initialization rather than as a check deeper in the code.